### DISCUSSION TITLE
Pytorch ignite compatibility

### DISCUSSION THREAD

**Discussion by scarpma** [pyg-team/pytorch_geometric#4454] [Q&A]
Hello everyone. I recently started using [pytorch ignite](https://pytorch.org/ignite/index.html) to train my non-graph-based NN. I think it is a great tool. Basically, it is a tool similar to PyTorch lighting.

I was wondering if there are any known problems for which PyTorch ignite should not work with PyTorch geometric.

Thanks in advance!

**Comment by rusty1s** [comment]
I don‘t think there should exist any major issues with combining PyG and Ignite. Happy to work on an example with you to showcase this as well. Just let me know :)

**Comment by scarpma** [comment]
Hello @rusty1s ! I'm happy to hear from you. I've started working on this and it seems to work all just fine. I would be happy to do that. One minor problem is that standard ignite trainer are built for models that accept one signle argument as `y_pred = model(x)`. In the pyg context this is often not valid, so I solved the problem by designing my models to accept the whole `Data` class, instead of signle tensors. I see that this is not the code style of the "official implemented" models.

Do you think there's a better work around ? I don't think that manually designing the `update_step`  function inside the PyTorch Ignite trainer would be a good idea (in order to accept models with multiple inputs).

**Reply by rusty1s** [reply]
I am not really familiar with PyTorch Ignite, but passing the whole data object to forward seems fine to me. As an alternative, you may want to use named tuples, which makes the required arguments passed to forward more explicit.

**Reply by scarpma** [reply]
Could you show me a link of a tutorial to take inspiration from (for style etc.) ?

**Reply by rusty1s** [reply]
How about we provide a simple example to begin with, similar to our [PL example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/pytorch_lightning/gin.py).

**Reply by scarpma** [reply]
I have the python script for the example. What should I do now ? Sorry but I'm not very familiar with contributing to open source.
Should I clone the whole repository and create a pull request ?

**Reply by rusty1s** [reply]
Yes, that would be ideal:
1. Fork PyG
2. Make and push changes in your fork
3. Create a PR on GitHub

Let me know if you have any trouble with any lf these steps.
